pkg/test/case.go

```diff
@@ -347,9 +349,11 @@ func (t *Case) Run(test *testing.T, ts *report.Testsuite) {
 		clients[testStep.Kubeconfig] = cl
 	}
 
-	for _, c := range clients {
-		if err := t.CreateNamespace(test, c, ns); err != nil {
-			setupReport.Failure = report.NewFailure(err.Error(), nil)
+	for kc, c := range clients {
+		if err = t.CreateNamespace(test, c, ns); apierrors.IsAlreadyExists(err) {
+			t.Logger.Logf("namespace %q already exists, using kubeconfig %q", ns.Name, kc)
```

Comment:
I'm having mixed feelings about this.

Before your change, encountering an existing namespace when trying to create one would abort the test and leave namespace alone.
After your change, it will silently assume ownership, and later delete the namespace.

The new behaviour is probably the expected one in a typical case, but could be a disaster if the user didn't mean to reuse (nor delete) the namespace.

I feel it should at least be documented.

WDYT?

Reply:
> After your change, it will silently assume ownership, and later delete the namespace.

Is there an ownership label that Kuttl adds to its resources? Ideally, the namespaces (and other resources) Kuttl is creating should only be cleaned if they have that label.

Reply:
That's a good idea. But I don't know the answer. I think I've seen something like that somewhere in the code, but not sure. Shouldn't be hard to find out experimentally, though. And if there is no such thing, we can add this.

Reply:
But probably in a different PR :-)

Reply:
> Before your change, encountering an existing namespace when trying to create one would abort the test and leave namespace alone. After your change, it will silently assume ownership, and later delete the namespace.

Looks like `CreateNamespace` is only called if the Namespace is supposed to be auto-created: https://github.com/kudobuilder/kuttl/blob/280792ae8ad0b4fb1b4387a2ef085ac8daabdf3f/pkg/test/case.go#L100-L103
and so Namespaces not created by Kuttl shouldn't be cleaned up:
https://github.com/kudobuilder/kuttl/blob/280792ae8ad0b4fb1b4387a2ef085ac8daabdf3f/pkg/test/case.go#L113-L119

> The new behaviour is probably the expected one in a typical case, but could be a disaster if the user didn't mean to reuse (nor delete) the namespace.
> 
> I feel it should at least be documented.

I think `CreateNamespace` ensures that we're not deleting Namespaces Kuttl didn't create and if the user wants to keep it, they can specify `SkipDelete` in the `TestSuite`.

Reply:
Ah, nice catch!